Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -189,7 +189,7 @@ public final class CacheAdvance<T: Codable> {
     /// - Parameter messageLength: the length of the next message that will be written.
     private func prepareReaderForWriting(dataOfLength messageLength: Bytes) throws {
         while writer.offsetInFile < reader.offsetInFile
-            && reader.offsetInFile < writer.offsetInFile + messageLength
+            && reader.offsetInFile <= writer.offsetInFile + messageLength
```

Comment:
Is there a way we can add a test to directly check for this?

Reply:
Good point. I should write a test for this. It's not the easiest of conditions to replicate. But will give it a shot.